docs/api.md

```diff
@@ -1,386 +1,840 @@
-  * `userID`: (via URL path) the ID of the user to fetch. The user has to exist.
+```js
+DELETE /api/emotes/package
 
-Returns `{user}` if successful, where `user` is a [user object](#user-objects) corresponding to the user with the given ID.
+<- {}
+```
 
-### GET `/api/user-list`
+---
 
-Returns `{users}`, where `users` is an array of every registered user on the server as [user objects](#user-objects).
+## Sessions
 
-### GET `/api/username-available/:username`
+Model:
+```js
+{
+  "id": string,
+  "dateCreated": number,
+}
+```
 
-- Parameters:
-  * `username`: (via URL path) the username to check.
+<a name='get-sessions'></a>
+### Fetch the current user's sessions [GET /api/sessions]
++ requires session
 
-Returns `{available}`, where `available` is a boolean set to whether or not the given username is available. This endpoint is handy when making a registration form (which might automatically check if a username is available before sending the [register](#post-apiregister) request).
+Responds with `{ sessions }`, where `sessions` is an array of [sessions](#sessions) that also represent the user that the provided session represents (the callee; you).
 
-### POST `/api/login`
+```js
+GET /api/sessions
 
-- Parameters:
-  * `username`: (via body; string) the username to log in as. There must be a user with this name.
-  * `password`: (via body; string) the password to use. This must (when hashed) match the user's password.
-- [Authentication](#authentication): never required.
+<- {
+<-   "sessions": [
+<-     "id": "12345678-ABCDEFGH",
+<-     "dateCreated": 123456789000
+<-   ]
+<- }
+```
 
-Attempts to log in as a user, creating a new session. Returns `{sessionID}` if successful, where `sessionID` is the ID of the newly-created session.
+<a name='login'></a>
+### Login [POST /api/sessions]
++ never requires session
++ `username` (string)
++ `password` (string)
 
-### GET `/api/session/:sessionID`
+Responds with `{ sessionID }` if successful, where `sessionID` is the ID of the newly-created session. Related endpoint: [register](#register).
 
-- Parameters:
-  * `sessionID`: (via URL path) the session ID to fetch. The session must exist.
-- [Authentication](#authentication): never required. Should, however, be provided in the URL.
+```js
+POST /api/sessions
 
-Returns `{session: {id, dateCreated, user}}` if successful, where `user` is a [user object](#user-object) of the user which the session represents. This endpoint is useful when grabbing information about the logged in user (e.g. at the startup of a client program, which may display the logged in user's username in a status bar). Does not require [authorization](#authorization).
+-> {
+->   "username": "admin",
+->   "password": "abcdef"
+-> }
 
-### POST `/api/delete-sessions`
+<- {
+<-   "sessionID": "12345678-ABCDEFGH"
+<- }
+```
 
-- Parameters:
-  * `sessionIDs`: (via body; array of strings) the IDs of the sessions to be deleted.
+### Fetch session details [GET /api/session/:id]
```

Comment:
I'm changing this to `/api/sessions/:id`, by the way, to match `/api/users/:id`, `api/channels/:id`, etc.